src/btree/row_modify.c

```diff
@@ -372,9 +375,21 @@ __wt_update_obsolete_check(
             if (size != 0)
                 __wt_cache_page_inmem_decr(session, page, size);
         }
-        return (next);
     }
 
+    /*
+     * Force evict a page when there are more than WT_THOUSAND updates to a single item. WT_THOUSAND
```

Comment:
A few grammatical issues in the second sentence, maybe reword as:

"Increasing the minSnapshotHistoryWindowInSeconds to 300 introduced a performance regression in which the average number of updates on a single item was approximately 1000 in write-heavy workloads. This is why we use WT_THOUSAND here."

Reply:
Fixed